test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsQueueResourceTest.kt

```diff
@@ -76,16 +77,36 @@ class HmppsQueueResourceTest : IntegrationTestBase() {
       verify(messageServiceSpy).handleMessage("message1")
       verify(messageServiceSpy).handleMessage("message2")
     }
+
+    @Test
+    fun `should transfer messages from another DLQ to another queue and process them`() {
+      anotherSqsDlqClient.sendMessage(anotherDlqUrl, "message3")
+      anotherSqsDlqClient.sendMessage(anotherDlqUrl, "message4")
+      await untilCallTo { anotherSqsDlqClient.countMessagesOnQueue(anotherDlqUrl) } matches { it == 2 }
+
+      webTestClient.put()
+        .uri("/queue-admin/retry-dlq/${hmppsQueueProperties.anotherQueue().dlqName}")
+        .headers { it.authToken() }
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus().isOk
+
+      await untilCallTo { anotherSqsDlqClient.countMessagesOnQueue(anotherDlqUrl) } matches { it == 0 }
+      await untilCallTo { anotherSqsClient.countMessagesOnQueue(anotherQueueUrl) } matches { it == 0 }
+
+      verify(anotherMessageServiceSpy).handleMessage("message3")
+      verify(anotherMessageServiceSpy).handleMessage("message4")
+    }
   }
 
   @Nested
   inner class RetryAllDlqs {
-    // TODO when we start handling multiple queues add another queue/dlq combination to this test
     @Test
     fun `should transfer messages from DLQ to main queue and process them`() {
       sqsDlqClient.sendMessage(dlqUrl, "message1")
-      sqsDlqClient.sendMessage(dlqUrl, "message2")
-      await untilCallTo { sqsDlqClient.countMessagesOnQueue(dlqUrl) } matches { it == 2 }
+      anotherSqsDlqClient.sendMessage(anotherDlqUrl, "message2")
+      await untilCallTo { sqsDlqClient.countMessagesOnQueue(dlqUrl) } matches { it == 1 }
```

Comment:
Now adds a message to each dlq to prove that all dlqs are processed by the `/retry-dlqs` endpoint.